lib/lte_link_control/lte_lc.c

```diff
@@ -1036,99 +1038,120 @@ int lte_lc_psm_get(int *tau, int *active_time)
 		return -EINVAL;
 	}
 
+	ptw_param = (mode == LTE_LC_LTE_MODE_LTEM) ? ptw_param_ltem :
+						     ptw_param_nbiot;
+
 	if (ptw != NULL) {
 		strcpy(ptw_param, ptw);
-		LOG_DBG("PTW set to %s", log_strdup(ptw_param));
+		LOG_DBG("PTW set to %s for %s", log_strdup(ptw_param),
+			(mode == LTE_LC_LTE_MODE_LTEM) ? "LTE-M" : "NB-IoT");
 	} else {
 		*ptw_param = '\0';
-		LOG_DBG("PTW use default");
+		LOG_DBG("PTW use default for %s",
+			(mode == LTE_LC_LTE_MODE_LTEM) ? "LTE-M" : "NB-IoT");
 	}
 
 	return 0;
 }
 
 int lte_lc_edrx_req(bool enable)
 {
-	int err, actt;
+	int err;
+	int actt[] = {AT_CEDRXS_ACTT_WB, AT_CEDRXS_ACTT_NB};
 	char req[25];
 
-	if (sys_mode_current == LTE_LC_SYSTEM_MODE_NONE) {
-		err = lte_lc_system_mode_get(&sys_mode_current, NULL);
-		if (err) {
-			return err;
-		}
-	}
-
-	switch (sys_mode_current) {
-	case LTE_LC_SYSTEM_MODE_LTEM:
-	case LTE_LC_SYSTEM_MODE_LTEM_GPS:
-		actt = AT_CEDRXS_ACTT_WB;
-		break;
-	case LTE_LC_SYSTEM_MODE_NBIOT:
-	case LTE_LC_SYSTEM_MODE_NBIOT_GPS:
-		actt = AT_CEDRXS_ACTT_NB;
-		break;
-	default:
-		LOG_ERR("Cannot request eDRX for this system mode (%d)",
-			sys_mode_current);
-		return -EOPNOTSUPP;
-	}
-
 	if (enable) {
-		if (strlen(edrx_param) == 4) {
-			snprintf(req, sizeof(req),
-				"AT+CEDRXS=2,%d,\"%s\"", actt, edrx_param);
-		} else {
-			snprintf(req, sizeof(req),
-				"AT+CEDRXS=2,%d", actt);
+		for (size_t i = 0; i < ARRAY_SIZE(actt); i++) {
+			char *edrx_param = (actt[i] == AT_CEDRXS_ACTT_WB) ?
+						edrx_param_ltem :
+						edrx_param_nbiot;
+
+			if (strlen(edrx_param) == 4) {
+				snprintk(req, sizeof(req),
+					 "AT+CEDRXS=2,%d,\"%s\"",
+					 actt[i], edrx_param);
+			} else {
+				snprintk(req, sizeof(req), "AT+CEDRXS=2,%d", actt[i]);
+			}
+
+			err = at_cmd_write(req, NULL, 0, NULL);
+			if (err) {
+				LOG_ERR("Failed to enable eDRX, error: %d", err);
+				return err;
+			}
 		}
-		err = at_cmd_write(req, NULL, 0, NULL);
 	} else {
 		err = at_cmd_write(edrx_disable, NULL, 0, NULL);
-	}
-	if (err) {
-		LOG_ERR("Failed to %s eDRX, error: %d",
-			enable ? "enable" : "disable", err);
-		return err;
+		if (err) {
+			LOG_ERR("Failed to disable eDRX, error: %d", err);
+			return err;
+		}
 	}
 
 	/* PTW must be requested after eDRX is enabled */
 	if (enable) {
-		if (strlen(ptw_param) == 4) {
-			snprintf(req, sizeof(req),
-				"AT%%XPTW=%d,\"%s\"", actt, ptw_param);
-		} else {
-			snprintf(req, sizeof(req),
-				"AT%%XPTW=%d", actt);
-		}
-		err = at_cmd_write(req, NULL, 0, NULL);
-		if (err) {
-			LOG_ERR("Failed to request PTW (%s), error: %d",
-				log_strdup(req), err);
-			return err;
+		for (size_t i = 0; i < ARRAY_SIZE(actt); i++) {
+			char *ptw_param = (actt[i] == AT_CEDRXS_ACTT_WB) ?
+						ptw_param_ltem :
+						ptw_param_nbiot;
+
+			if (strlen(ptw_param) == 4) {
+				snprintk(req, sizeof(req),
+					"AT%%XPTW=%d,\"%s\"",
+					actt[i], ptw_param);
+			} else {
+				snprintk(req, sizeof(req),
+					"AT%%XPTW=%d", actt[i]);
```

Comment:
This should not be needed because setting +CEDRXS sets the PTW value as default automatically. But if the code would be removed, there should probably still be a comment that PTW is reset automatically.

Setting the PTW could also be moved inside the "`if (enable) {`" for eDRX.

I leave the choice up to you. :)